### PR TITLE
fix: get/set Appmap state

### PR DESCRIPTION
**Pull request by sashahonchar**
This functionality was added in #169 , but later it was removed because there was a bug with not working `setState` command when few Appmaps were opened.

### CODE REVIEW

**Comment by kgilpin** [comment]
Will it be easy to go to a specific event in the Trace view? If I know the
event ID?

**Comment by sashahonchar** [comment]
I think we should create new command for that @kgilpin

**Comment by dustinbyrne** [comment]
Don't we support that already through the `currentView` key?

File: src/scenarioViewer.ts [hunk 1]
```diff
@@ -32,6 +33,8 @@ export class ScenarioProvider implements vscode.CustomTextEditorProvider {
     webviewPanel: vscode.WebviewPanel
     /* _token: vscode.CancellationToken */
   ): Promise<void> {
+    ScenarioProvider.currentWebView = webviewPanel;
```
Comment (dustinbyrne) [inline src/scenarioViewer.ts:36]: Is this also executed when switching tabs?

Comment (sashahonchar) [reply src/scenarioViewer.ts:36]: No, It's not executed when you switch between 2 opened appmap viewers in VSCode

Comment (dustinbyrne) [reply src/scenarioViewer.ts:36]: Is there a way to listen for that event? Or can we avoid the issue by accepting two arguments instead:
- a path to an AppMap file to be opened by the command
- a state string to be immediately loaded upon opening

Comment (sashahonchar) [reply src/scenarioViewer.ts]: @dustinbyrne , I reviewed VSCode docs and found a solution. https://github.com/applandinc/vscode-appland/pull/232/commits/427e42761d12cb2e556a341ffe1bb5100e31d439

**Comment by kgilpin** [comment]
@sashahonchar is this branch still alive?

I need a command to open the trace view to a specified event.

**Comment by sashahonchar** [comment]
> @sashahonchar is this branch still alive?
> 
> I need a command to open the trace view to a specified event.

@kgilpin , @dustinbyrne  I think this branch should be closed and new issue with detailed description should be opened.

**Comment by sashahonchar** [comment]
@kgilpin, Yeah, I just rebased this branch from master.

**Review by dustinbyrne — APPROVED** [review]
I'm happy to merge this as-is, but I think to better support Kevin's use-case, the command should take both a file and state string. Then, once executed, will open the file with the given state. I'll open an issue.

**Comment by dustinbyrne** [comment]
https://github.com/applandinc/vscode-appland/issues/336

File: src/extension.ts [hunk 2]
```diff
@@ -86,6 +86,33 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
...
+            }
+
+            if (queryParams.get('state')) {
+              // let's give some time to open AppMap file before setting state
+              setTimeout(() => {
+                vscode.commands.executeCommand(
+                  'appmap.setAppmapStateNoPrompt',
+                  queryParams.get('state')
+                );
+              }, 1000);
```
Comment (dustinbyrne) [inline src/extension.ts:109]: This `setTimeout` seems fragile. Is it possible to send this state as part of the command to open the file?

Comment (sashahonchar) [reply src/extension.ts]: Good point!
Now we gonna store it in `globalState` and apply after webview load. https://github.com/applandinc/vscode-appland/pull/232/commits/b618f57285e43d50cffd5a25a4f9bcaa539aba0e

**Comment by dustinbyrne** [comment]
I've looked at the code, but I still need to try running this myself. I'll have a chance tomorrow.

**Pull request merged**